Migrate command: --write-sql default to false

sserbin opened the pull request:
Set default of write-sql to false to mimic behavior of doctrine/migrations (see https://github.com/doctrine/migrations/blob/5713b45c933122e509d9b31c767b420c3dfed399/lib/Doctrine/Migrations/Tools/Console/Command/MigrateCommand.php#L54)

Previously it defaulted to `null` which made the doctrine (upstream) command defaulting to getcwd() (via null-coalesce) and always generating an sql dump

sserbin committed:
Migrate command: --write-sql default to false

Set default of write-sql to false to mimic behavior of doctrine/migrations (see https://github.com/doctrine/migrations/blob/5713b45c933122e509d9b31c767b420c3dfed399/lib/Doctrine/Migrations/Tools/Console/Command/MigrateCommand.php#L54)

Previously it defaulted to `null` which made the doctrine (upstream)
command defaulting to getcwd() (via null-coalesce) and always generating an sql dump

dpslwk commented:
Execute command might also want the same change

sserbin committed:
set default=false for write-sql for execute command

sserbin commented:
> Execute command might also want the same change

@dpslwk yep, added :+1:

eigan commented:
Not possible to set it in the signature as `--write-sql=false :`? Having it in `handle` will hide default value from `--help`.

eigan committed:
Test did not notice difference between false and NULL

eigan commented:
Ref my previous comment: not possible to set `false` (bool) default in `$signature`, but moving it to __construct will result in `--help` show the proper default value. 

I fixed the testcase which was supposed to detect these issues. Move your code to __construct and the tests should pass :)

sserbin committed:
set default value for diff:check-database-platform and execute:write-sql

also set the defaults in constructor (as @eigan pointed out) for
defaults values to be displayed when running --help

sserbin commented:
> 

thanks, I've also updated default value for `diff --check-database-platform` and `execute --write-sql` as tests for those were failing as well. Should all be green now

eigan commented:
Thanks @sserbin ! 🙂
